File: TEAM B2 (Season 2)/Week 3/Musk/[해시_폰켓몬].cpp
#include <vector>
#include <unordered_set>
using namespace std;

// https://velog.io/@yuiop1029/프로그래머스-폰켓몬

int solution(vector<int> nums) {
    
    unordered_set<int> set;
    
    for(auto item : nums ) {
        if(!(set.find(item) != set.end())) { set.insert(item); }
    }
    
    int answer = set.size() > nums.size()/2 ? nums.size()/2 : set.size();
    return answer;
}

File: TEAM B2 (Season 2)/Week 3/Moana/[해시_폰켓몬].py
def solution(nums):
    tmpnum = max(nums)+1
    pocketmonList = [0 for i in range(tmpnum)]
    answer = 0
    
    for i in nums:
        pocketmonList[i] += 1
        
    for i in pocketmonList:
        if i != 0:
            answer += 1
        if answer >= len(nums)/2:
            break
    
    return answer